contracts/cw-proposal-single/src/contract.rs

```diff
@@ -714,15 +729,23 @@ pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response,
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
-    if msg.id % 2 == 0 {
-        // Proposal hook so we can just divide by two for index
-        let idx = msg.id / 2;
-        PROPOSAL_HOOKS.remove_hook_by_index(deps.storage, idx)?;
-        Ok(Response::new())
-    } else {
-        // Vote hook so we can minus one then divid by two for index
-        let idx = (msg.id - 1) / 2;
-        VOTE_HOOKS.remove_hook_by_index(deps.storage, idx)?;
-        Ok(Response::new())
+    let repl = TaggedReplyId::new(msg.id);
+    match repl {
+        TaggedReplyId::FailedProposalExecution(proposal_id) => {
+            let mut prop = PROPOSALS
+                .may_load(deps.storage, proposal_id)?
+                .ok_or(ContractError::NoSuchProposal { id: proposal_id })?;
+            prop.status = Status::ExecutionFailed;
+            PROPOSALS.save(deps.storage, proposal_id, &prop)?;
```

Comment:
Instead of doing `may_load` followed by `save` `cosmwasm_std::Map` has an `update` method for these situations.

Example of doing something similar: https://github.com/DA0-DA0/dao-contracts/blob/6dc92d9cba7f00d570f97fbceadd334fb61d6ccf/contracts/cw-proposal-single/src/contract.rs#L319-L346

Reply:
Can do the same in `cw-proposal-multiple`.

Reply:
Yep, forgot about it for some reason)
Thanks.